docs/design/ovs-pipeline.md

```diff
@@ -54,525 +54,605 @@
+    As a consequence, packets from the tunnel should be seamlessly forwarded to table [UnSNAT].
+- Flows 4-7 match packets from local Pods.
+  - Match condition `in_port=<OVS_PORT_NAME>` is to match packets which are received on local Pod ports.
+  - Action `set_field:0x3/0xf->reg0` is to load `FromLocalRegMark` (bits 0-4 of `NXM_NX_REG0`, `PktSourceField` is 0x3)
+    to mark packet source.
+  - Action `goto_table:SpoofGuard` is the same as flow 1.
+- Flow 8 is the default drop flow, rarely used in practice.
+
+### SpoofGuard
+
+This table prevents IP [spoofing](https://en.wikipedia.org/wiki/Spoofing_attack) from local Pods. For each Pod (as identified by the ingress port), we ensure that
+the source IP and MAC addresses are correct, i.e. match the values configured on the interface when Antrea set-up
+networking for the Pod.
 
-This table handles "tracked" packets (packets which are moved to the tracked
-state by the previous table [ConntrackTable]) and "untracked" packets (packets
-is not in tracked state).
+If you dump the flows for this table, you may see the following:
 
-This table serves the following purposes:
+```text
+1. cookie=0x2010000000000, table=SpoofGuard, priority=200,ip,in_port="antrea-gw0" actions=goto_table:UnSNAT
+2. cookie=0x2010000000000, table=SpoofGuard, priority=200,ip,in_port="coredns--da1990",dl_src=ee:76:de:a8:dc:c9,nw_src=10.10.0.2 actions=goto_table:UnSNAT
+3. cookie=0x2010000000000, table=SpoofGuard, priority=200,ip,in_port="coredns--bb7794",dl_src=a6:ef:11:a7:87:b9,nw_src=10.10.0.3 actions=goto_table:UnSNAT
+4. cookie=0x2010000000000, table=SpoofGuard, priority=200,ip,in_port="nginx-d9-cd1533",dl_src=2e:ba:06:b2:44:91,nw_src=10.10.0.7 actions=goto_table:UnSNAT
+5. cookie=0x2010000000000, table=SpoofGuard, priority=200,ip,in_port="nginx-d9-b93cc5",dl_src=c2:5a:5e:50:95:9b,nw_src=10.10.0.8 actions=goto_table:UnSNAT
+6. cookie=0x2000000000000, table=SpoofGuard, priority=0 actions=drop
+```
 
-* For tracked Service packets, bit 19 in NXM_NX_REG0 will be set to 0x1, then
-  the tracked packet will be forwarded to [EgressRuleTable] directly.
-* Drop packets reported as invalid by conntrack.
-* Non-Service tracked packets goes to [EgressRuleTable] directly.
-* Untracked packets goes to [SessionAffinityTable] and [ServiceLBTable].
+- Flow 1 matches packets from local Antrea gateway, but does not check source IP. There are some cases where the source
+  IP of packets through local Antrea gateway is not the local Antrea gateway IP:
+  - When Antrea is deployed with kube-proxy and AntreaProxy is not enabled, packets from local Pods destined to Services
+    will first go through the gateway, get load-balanced by the kube-proxy datapath (DNAT) then sent back through the
+    gateway. This means that legitimate packets can be received on the gateway port with a source IP belonging to a local
+    Pod.
+  - When both AntreaProxy and proxyAll are enabled, packets from external destined to Services will be routed to OVS
```

Comment:
In case 2, packets are forwarded by iptables and iptsets installed by AntreaProxy to antrea-gw0, but in case 3, packets are load-balanced by the kube-proxy and forwarded to antrea-gw0.